test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -14,47 +13,85 @@ namespace Microsoft.ML.Tests.Transformers
                 separator: ';', hasHeader: true)
-                .Read(dataSource);
+                .Read(new MultiFileSource(_dataSource));
 
-            var invalidData = TextLoader.CreateReader(env,
-                c => (label: c.LoadFloat(11), features: c.LoadText(0, 10)),
+            var invalidData = TextLoader.CreateReader(_env,
+                c => (label: c.LoadFloat(11), weight: c.LoadFloat(0), features: c.LoadText(1, 10)),
                 separator: ';', hasHeader: true)
-                .Read(dataSource);
+                .Read(new MultiFileSource(_dataSource));
+
+            var est = new PcaEstimator(_env, "features", "pca", rank: 4, seed: 10);
+            TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
 
-            var est = new PcaEstimator(env, "features", "pca", rank: 5, advancedSettings: s => {
-                    s.Seed = 1;
-                });
+            var est_non_default_args = new PcaEstimator(_env, "features", "pca", rank: 3, weightColumn: "weight", overSampling: 2, center: false);
+            TestEstimatorCore(est_non_default_args, data.AsDynamic, invalidInput: invalidData.AsDynamic);
 
-            // The following call fails because of the following issue
-            // https://github.com/dotnet/machinelearning/issues/969
-            // TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
+            Done();
+        }
 
+        [Fact]
+        public void TestPcaEstimator()
+        {
+            var data = TextLoader.CreateReader(_env,
+                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+                separator: ';', hasHeader: true)
+                .Read(new MultiFileSource(_dataSource));
+
+            var est = new PcaEstimator(_env, "features", "pca", rank: 5, seed: 1);
             var outputPath = GetOutputPath("PCA", "pca.tsv");
-            using (var ch = env.Start("save"))
+            using (var ch = _env.Start("save"))
             {
-                var saver = new TextSaver(env, new TextSaver.Arguments { Silent = true, OutputHeader = false });
-                IDataView savedData = TakeFilter.Create(env, est.Fit(data.AsDynamic).Transform(data.AsDynamic), 4);
-                savedData = new ChooseColumnsTransform(env, savedData, "pca");
+                IDataView savedData = TakeFilter.Create(_env, est.Fit(data.AsDynamic).Transform(data.AsDynamic), 4);
+                savedData = new ChooseColumnsTransform(_env, savedData, "pca");
 
                 using (var fs = File.Create(outputPath))
-                    DataSaverUtils.SaveDataView(ch, saver, savedData, fs, keepHidden: true);
+                    DataSaverUtils.SaveDataView(ch, _saver, savedData, fs, keepHidden: true);
             }
 
             CheckEquality("PCA", "pca.tsv");
             Done();
         }
+
+        [Fact]
+        public void TestPcaPigsty()
```

Comment:
>Pigsty [](start = 27, length = 6)

1. Rename to `TestPcaStatic`: we don't want codename `Pigsty` in the codebase
2. Move to `StaticPipelineTests`: we group pigsty tests together,  and non-pigsty ones together

Reply:
Moved. 

---
In reply to: [227067279](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227067279) [](ancestors = 227067279)